frontend/src/javascript/codeParser/CodeParserUtils.js

```diff
@@ -83,20 +85,26 @@ export class CodeParserUtils {
     }
 
     static insertNecessaryIncludes(code) {
-        const regex = /#include[ \t]*<iostream>/g;
+        let regex = /#include[ \t]*<iostream>/g;
```

Comment:
Don't repeat yourself, jeżeli dobrze widzę, to idealnie to się nadaje na osobną funkcję, która dodaje odpowiednie nagłówki jeżeli ich nie ma (mogła by nawet przyjmować w argumencie tablicę tych nagłówków i się po niej iterowała.